hudi-gcp/src/main/java/org/apache/hudi/gcp/bigquery/HoodieBigQuerySyncClient.java

```diff
@@ -310,6 +311,16 @@ public boolean tableNotExistsOrDoesNotMatchSpecification(String tableName) {
     boolean manifestDoesNotExist =
         externalTableDefinition.getSourceUris() == null
             || externalTableDefinition.getSourceUris().stream().noneMatch(uri -> uri.contains(ManifestFileWriter.ABSOLUTE_PATH_MANIFEST_FOLDER_NAME));
+    String basePathInTableDefinition = externalTableDefinition.getHivePartitioningOptions() == null ? "" :
+        externalTableDefinition.getHivePartitioningOptions().getSourceUriPrefix();
+    // remove trailing slash
+    basePathInTableDefinition = org.apache.commons.lang3.StringUtils.stripEnd(basePathInTableDefinition, "/");
+    basePath = org.apache.commons.lang3.StringUtils.stripEnd(basePath, "/");
```

Comment:
Avoid using `org.apache.commons.lang3.StringUtils` class to avoid possible dependency conflict.  Could you add a util `stripEnd` in `org.apache.hudi.common.util.StringUtils`?